pkg/handler/v2/external_cluster/gke.go

```diff
@@ -192,3 +209,65 @@ func deleteGKENodePool(ctx context.Context, cluster *kubermaticapiv1.ExternalClu
 	_, err = req.Context(ctx).Do()
 	return err
 }
+
+func patchGKEMachineDeployment(ctx context.Context, old, new *apiv2.ExternalClusterMachineDeployment, cluster *kubermaticapiv1.ExternalCluster, secretKeySelector provider.SecretKeySelectorValueFunc, credentialsReference *providerconfig.GlobalSecretKeySelector, clusterProvider provider.ExternalClusterProvider) (*apiv2.ExternalClusterMachineDeployment, error) {
+	sa, err := secretKeySelector(credentialsReference, resources.GCPServiceAccount)
+	if err != nil {
+		return nil, err
+	}
+	svc, project, err := gcp.ConnectToContainerService(sa)
+	if err != nil {
+		return nil, err
+	}
+
+	// The GKE can update Node Pool size or NodePoll object. Can't change both because first update blocks the second one.
+	// It's required to update Node Poll size separately.
+
+	// only when size was updates otherwise change NodePoll object with other parameters
+	if old.Spec.Replicas != new.Spec.Replicas {
+		sizeRequest := &container.SetNodePoolSizeRequest{
+			NodeCount: int64(new.Spec.Replicas),
+		}
+		sizeReq := svc.Projects.Zones.Clusters.NodePools.SetSize(project, cluster.Spec.CloudSpec.GKE.Zone, cluster.Spec.CloudSpec.GKE.Name, old.Name, sizeRequest)
+		_, err = sizeReq.Context(ctx).Do()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Can you before returning new here update this as new has both replica and version patch `new.NodeDeployment.Spec.Template.Versions.Kubelet = old.Spec.Template.Versions.Kubelet`